lib/rules/no-component-lifecycle-hooks.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+
+const { isComponentLifecycleHookName } = emberUtils;
+
+const ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS =
+  'Do not use classic ember components lifecycle hooks. Prefer using "@ember/render-modifiers" or custom functional modifiers.';
+
+module.exports = {
+  ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS,
+  meta: {
+    docs: {
+      description:
+        'Prevents usage of "classic" ember component lifecycle hooks. Render modifiers or custom functional modifiers should be used instead.',
+      category: 'Ember Octane',
+      recommended: false,
+      octane: true,
+    },
+    fixable: null, // or "code" or "whitespace"
+  },
+
+  create(context) {
+    let isInsideEmberComponent = false;
+
+    const isAClassicLifecycleHook = keyName => {
+      return isComponentLifecycleHookName(keyName) && isInsideEmberComponent;
+    };
+
+    const report = (context, node) => {
+      context.report(node, ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS);
+    };
+
+    return {
+      ClassDeclaration(node) {
+        isInsideEmberComponent =
+          emberUtils.isEmberComponent(context, node) ||
+          emberUtils.isGlimmerComponent(context, node);
+      },
+
+      CallExpression(node) {
+        isInsideEmberComponent = emberUtils.isEmberComponent(context, node);
+      },
+
+      'ClassDeclaration:exit'(node) {
+        isInsideEmberComponent = false;
+      },
+
+      'CallExpression:exit'(node) {
```

Comment:
This isn't quite right. There can be many `CallExpression`s inside a class so exiting one doesn't mean you actually left the class. You will actually want to store the current class node so you can tell when you are leaving it like in this example: https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/rules/no-get.js#L59-L66

Reply:
right, will fix that 👍

Reply:
fixed and added a test case